llvm/lib/Transforms/Scalar/LoopDistribute.cpp

```diff
@@ -521,6 +521,21 @@ class InstPartitionContainer {
           Partition = -1;
       }
       assert(Partition != -2 && "Pointer not belonging to any partition");
+      // All the store context uses of our address were processed,
+      // Now make sure we don't have cross partition loads.
+      if (RtPtrCheck->Pointers[I].IsWritePtr) {
+        if (Ptr->hasOneUse() || Partition == -1)
+          continue;
+
+        for (User *U : Ptr->users())
+          if (auto *CurLoad = dyn_cast<LoadInst>(U))
+            if (L->contains(CurLoad->getParent()))
+              if (Partition != (int)this->InstToPartitionId[CurLoad]) {
```

Comment:
```suggestion
          auto *CurLoad = dyn_cast<LoadInst>(U);
          if (!CurLoad)
            continue; 
          if (!L->contains(CurLoad->getParent()))
            continue;
          if (Partition == (int)this->InstToPartitionId[CurLoad]) 
            continue;
```
[Use Early Exits and continue to Simplify Code](https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code)